src/components/search-form/search-form.tsx

```diff
@@ -0,0 +1,25 @@
+import styles from '../movie-list-page/styles.module.scss'
+import { Search } from '../../ui'
+import { useSearchParams } from 'react-router-dom'
+import { useCallback } from 'react'
+
+export const SearchForm = () => {
+  const [searchQuery, setSearchQuery] = useSearchParams()
+  const handleSearch = useCallback(
+    (query: string) => {
+      searchQuery.set('query', query)
+      setSearchQuery(searchQuery)
```

Comment:
It's better to use a function for setting new search params:

```suggestion
      setSearchQuery((prev) => {
        prev.set('query', query);
        return prev;
      });
```

This way you will get rid of dependency on `searchQuery` for your `useCallback`.